packages/ui/src/components/filters/filters-bar/filter-variants/number.tsx

```diff
@@ -47,32 +46,30 @@ const Number = ({ filter, onUpdateFilter }: NumberFilterProps) => {
   }
 
   return (
-    <div className="flex items-center px-2 pb-2.5">
-      <DropdownMenuItem className="relative w-full p-0">
-        <Input
-          className="w-full"
-          type="text"
-          value={value}
-          placeholder="Type a value..."
-          onChange={handleInputChange}
+    <div className="flex items-center px-2 pb-2.5 relative">
+      <Input
+        className="w-full"
+        type="text"
+        value={value}
+        placeholder="Type a value..."
+        onChange={handleInputChange}
+        onClick={e => {
+          e.stopPropagation()
+          e.preventDefault()
+        }}
+      />
+
+      {value && (
+        <button
```

Comment:
Since this is an icon-only button, can we add an `aria-label` to help describe it for screen readers?
```suggestion
        <button
          aria-label="clear filter"
```